crates/jit/src/instantiate.rs

```diff
@@ -421,6 +431,19 @@ impl CompiledModule {
             None => bincode::deserialize(section(ELF_WASMTIME_INFO)?)
                 .context("failed to deserialize wasmtime module info")?,
         };
+        let is_branch_protection_enabled = info.meta.is_branch_protection_enabled;
+
+        if remap && is_branch_protection_enabled {
+            // TODO: Get the correct mapping the first time.
```

Comment:
@alexcrichton I am not too happy about this part, but I tried separating the parsing of the ELF image and the changing of the memory protection done by `CodeMemory::publish()` and I ran into various borrow checker and lifetime issues that seemed beyond my Rust skills, so I went ahead with this suboptimal, but nevertheless correct solution.

Reply:
Oh dear, I'll work on improving this with some refactorings but otherwise this PR looks good to me, thanks for the updates!

Reply:
Ah I ended up with https://github.com/akirilov-arm/wasmtime/pull/1 which was easier than I thought

Reply:
Yeah, parsing the ELF header twice definitely helps.